resolver/resolver.cc

```diff
@@ -3298,8 +3298,16 @@ class ResolveSignaturesWalk {
         if (sig.seen.abstract) {
             method.data(ctx)->flags.isAbstract = true;
         }
-        if (sig.seen.incompatibleOverride) {
-            method.data(ctx)->flags.isIncompatibleOverride = true;
+        if (sig.seen.incompatibleOverride && sig.seen.incompatibleOverrideVisibility) {
+            if (auto e = ctx.beginError(exprLoc, core::errors::Resolver::InvalidMethodSignature)) {
```

Comment:
Could you add a test case for this?

Also, if both `override(allow_incompatible: false)` and `override(allow_incompatible: :visibility)` are used, we'll just fall into the `else if (sig.seen.incompatibleOverrideVisibility)` and report no errors right? I think that's okay behaviour, but probably worth adding a test case for that so that's documented.